JamJar/UINavigationController.swift

```diff
@@ -14,6 +14,12 @@ extension UINavigationController {
     }
     
     public override func shouldAutorotate() -> Bool {
-        return self.visibleViewController!.shouldAutorotate()
+        // If there's a visible viewcontroller, defer to that
+        if let visibleController = self.visibleViewController {
+            return visibleController.shouldAutorotate()
+        }
+        // Otherwise, don't autorotate???
+        // ETHAN: Is this what we want?
+        return false
```

Comment:
Yes it is. An entire app must be set to allow landscape and/or portrait. We wanted some views to allow landscape but not all. So we had to set the app to be all orientation friendly, and prevent autorotation actions on views by default unless we wanted landscape (since majority of the views are portrait only). If you think we shouldn't do this, or there is a better way, speak up.